components/engine/src/rocks/sst.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::sync::Arc;
+
+use super::util::get_fastest_supported_compression_type;
+use super::{ColumnFamilyOptions, DBCompressionType, Env, EnvOptions, ExternalSstFileInfo, DB};
+use crate::{CfName, CF_DEFAULT};
+use engine_rocksdb::SstFileWriter;
+
+/// A builder builds a SstWriter.
+pub struct SstWriterBuilder {
+    cf: Option<CfName>,
+    db: Option<Arc<DB>>,
+    in_memory: bool,
+}
+
+impl SstWriterBuilder {
+    /// Create a new SstWriterBuilder.
+    pub fn new() -> SstWriterBuilder {
+        SstWriterBuilder {
+            cf: None,
+            in_memory: false,
+            db: None,
+        }
+    }
+
+    /// Set DB for the builder. The builder may need some config from the DB.
+    pub fn set_db(mut self, db: Arc<DB>) -> Self {
+        self.db = Some(db);
+        self
+    }
+
+    /// Set CF for the builder. The builder may need some config from the CF.
+    pub fn set_cf(mut self, cf: CfName) -> Self {
+        self.cf = Some(cf);
+        self
+    }
+
+    /// Set it to true, the builder builds a in-memory SST builder.
+    pub fn set_in_memory(mut self, in_memory: bool) -> Self {
+        self.in_memory = in_memory;
+        self
+    }
+
+    /// Builder a SstWriter.
+    pub fn build(self, path: &str) -> Result<SstWriter, String> {
+        let mut env = None;
+        let mut io_options = if let Some(db) = self.db.as_ref() {
+            env = db.env();
+            let handle = db
+                .cf_handle(self.cf.unwrap_or(CF_DEFAULT))
+                .map_or_else(|| Err(format!("CF {:?} is not found", self.cf)), Ok)?;
```

Comment:
😕 If `cf_handle` returns a `Result` you could use `map_err` here, if `cf_handle` returns an `Option` you could use `ok_or_else` here